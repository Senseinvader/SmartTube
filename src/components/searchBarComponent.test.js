import React from 'react';
import configureStore from 'redux-mock-store';
import ConnectedSearchBarComponent, {SearchBarComponent} from './searchBarComponent';
import {shallow} from 'enzyme';
import toJson from "enzyme-to-json";

const mockStore = configureStore();
const mockOnSubmit = jest.fn(()=> console.log('tested'));
const initialState = {
    onSubmit: mockOnSubmit,
    searchReducer: {
        searchCriteria: ''
    }
};

const store = mockStore(initialState);

describe('Search bar component', () => {
    it('matches the snapshot', () => {
        const wrapper = shallow(<SearchBarComponent store={store}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
    it('has the form element', () => {
        const wrapper = shallow(<SearchBarComponent store={store}/>);
        const component = wrapper.dive();
        expect(component.find('form.form-container').length).toEqual(1);
    });
    describe('form', () => {
       it('calls handler function by clicking Sumbit', () => {
           const wrapper = shallow(<form onSubmit={mockOnSubmit}/>);
           wrapper.simulate('submit')
       });
    });
});
