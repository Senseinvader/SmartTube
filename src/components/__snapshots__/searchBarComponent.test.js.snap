// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component matches the snapshot 1`] = `
<div>
  <form
    className="form-container"
    onSubmit={[Function]}
  >
    <input
      className="search-input"
      placeholder="Search"
      type="text"
    />
    <input
      className="button-submit"
      type="submit"
      value="Search"
    />
  </form>
</div>
`;
